docs/android-installation.md

```diff
@@ -36,19 +36,18 @@ private static List<ReactPackage> getPackages() {
 4. Add permissionResult listener in `MainActivity.java`:
 
 ```java
-import io.wazo.callkeep.RNCallKeepModule; // Add this import line with others
+import io.wazo.callkeep.RNCallKeepModule; // Add these import lines
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
 public class MainActivity extends ReactActivity {
     // ...
 
     // Permission results
     @Override
-    public void onRequestPermissionsResult(int permsRequestCode, String[] permissions, int[] grantResults) {
-        super.onRequestPermissionsResult(permsRequestCode, permissions, grantResults);
-        switch (permsRequestCode) {
-            case RNCallKeepModule.REQUEST_READ_PHONE_STATE:
-                RNCallKeepModule.onRequestPermissionsResult(grantResults);
-                break;
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (grantResults.length > 0) {
```

Comment:
We're using [react-native-contacts](https://github.com/rt2zz/react-native-contacts) that also triggers `onRequestPermissionsResult`. Not using `RNCallKeepModule.REQUEST_READ_PHONE_STATE` here will invoque `RNCallKeepModule.onRequestPermissionsResult` when requesting for contacts permission (and other permission I guess).

This lead to a crash because `hasPhoneAccountPromise` is not yet defined :
```
07-24 15:12:12.216 32014 32014 E UncaughtException: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Promise.resolve(java.lang.Object)' on a null object reference
07-24 15:12:12.216 32014 32014 E UncaughtException: 	at io.wazo.callkeep.RNCallKeepModule.onRequestPermissionsResult(RNCallKeepModule.java:480)
07-24 15:12:12.216 32014 32014 E UncaughtException: 	at com.wazo.MainActivity.onRequestPermissionsResult(MainActivity.java:32)
07-24 15:12:12.216 32014 32014 E UncaughtException: 	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:7737)
```

Can you please put back the check on `RNCallKeepModule.REQUEST_READ_PHONE_STATE` ?

Reply:
Thanks, will correct this!